### DISCUSSION TITLE
Command palette option to run on all files

### DISCUSSION THREAD

**Discussion by xsrvmy** [microsoft/pylance-release#3542] [Enhancement]
One of the features pylance has is to run only on open files when editing to avoid performance issues. Is it possible to add a command to run pylance on the entire project in this case, without having to change settings?
Note that this is mostly a pylance and not a pyright issue, since with pyright using the cli is an option.

**Comment by judej** [comment]
If I understand this correctly, you're asking for a command to run diagnostics on the entire workspace. Is that right?

Moving this issue to discussion as an enhancement request for comments and upvotes.

**Reply by erictraut** [reply]
This is already exposed as a setting. If we were to add a palette option (or a command that appears in the command palette), it would simply be a shortcut for changing this setting. Is that what you have in mind?

Out of curiosity, why do you want to toggle this setting frequently? I tend to always leave it on "workspace" for my projects, and that works well.
